### ISSUE TITLE
NewType gives illegal annotation

### ISSUE THREAD

**Issue by RaderJason** [microsoft/pyright#5084]
Hello!
Using NewType raises "error: Illegal type annotation: variable not allowed unless it is a type alias" somewhat unexpectedly. Here is some reproducing code:

```python
from typing import Any, NewType

FooType = NewType("Foo", Any)

def bar(x: Any) -> FooType:
    return x
```
Similarly, if `FooType` is written as a type alias of the new type:
```python
from typing import Any, NewType, TypeAlias

FooType: TypeAlias = NewType("Foo", Any)

def bar(x: Any) -> FooType:
    return x
```
this raises "Invalid expression form for type alias definition." Note that `typing_extension` causes the same issue.

**Comment by erictraut** [comment]
You're using `NewType` in a manner in which it wasn't intended to be used. You're passing `Any` as the second argument to `NewType`. The `NewType` call creates a new type which acts as a subclass of the class you specify. The `Any` type isn't a class, so it cannot be subclassed. If your intent is to create a NewType that is a subtype of `object`, you should pass `object` as the second argument rather than `Any`.

That said, I agree that the error that pyright is emitting in this case is confusing.

I considered emitting an error if `Any` is passed to `NewType`, but I can see cases where that might occur legitimately — e.g. if you pass a class that was imported from a library whose import could not be resolved. I've therefore changed pyright's logic to handle this situation in a better manner.

This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.307, which I just published. It will also be included in a future release of pylance.
